File: WebSocket测试/代码/WebClientToolByWinForm/ConnectorObserverTextHandler.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using DotNetty.Buffers;
using FCARDIO.Core.Command;
using FCARDIO.Core.Connector;

namespace WebClientToolByWinForm
{
    /// <summary>
    /// 文本观察者
    /// </summary>
    public class ConnectorObserverTextHandler : INRequestHandle
    {
        private IObserverDebug _Debug;
        private System.Text.Encoding TextEncoding;


        public ConnectorObserverTextHandler(IObserverDebug log, Encoding enc)
        {
            _Debug = log;
            TextEncoding = enc;
        }


        public void Dispose()
        {
            return;
        }
        /// <summary>
        /// 接收的数据
        /// </summary>
        /// <param name="connector"></param>
        /// <param name="msg"></param>
        public void DisposeRequest(INConnector connector, IByteBuffer msg)
        {
            string sValue = msg.ReadString(msg.ReadableBytes, TextEncoding);

            _Debug.DisposeRequest(connector, sValue);
        }
        /// <summary>
        /// 发送的数据
        /// </summary>
        /// <param name="connector"></param>
        /// <param name="msg"></param>
        public void DisposeResponse(INConnector connector, IByteBuffer msg)
        {
            string sValue = msg.ReadString(msg.ReadableBytes, TextEncoding);

            _Debug.DisposeResponse(connector, sValue);
        }
    }

}

File: NetStandard/DoNetDrive.Core/DoNetDrive.Core/Connector/ObserverHandler/ConnectorObserverTextHandler.vb
Imports System.Text
Imports DotNetty.Buffers
Imports DoNetDrive.Core.Command

Namespace Connector
    ''' <summary>
    ''' 基于文本的通道观察者
    ''' </summary>
    Public Class ConnectorObserverTextHandler
        Implements INRequestHandle
        ''' <summary>
        ''' 输出接口
        ''' </summary>
        Private _Debug As IObserverTextDebug

        ''' <summary>
        ''' 文字编码
        ''' </summary>
        Private TextEncoding As Encoding


        Public Sub New(log As IObserverTextDebug, enc As Encoding)
            _Debug = log
            TextEncoding = enc
        End Sub

        ''' <summary>
        ''' 接收的数据
        ''' </summary>
        ''' <param name="connector"></param>
        ''' <param name="msg"></param>
        Public Sub DisposeRequest(connector As INConnector, msg As IByteBuffer) Implements INRequestHandle.DisposeRequest
            Dim sMsg = msg.ReadString(msg.ReadableBytes, TextEncoding)
            _Debug?.DisposeRequest(connector, sMsg)
        End Sub

        ''' <summary>
        ''' 发送的数据
        ''' </summary>
        ''' <param name="connector"></param>
        ''' <param name="msg"></param>
        Public Sub DisposeResponse(connector As INConnector, msg As IByteBuffer) Implements INRequestHandle.DisposeResponse
            Dim sMsg = msg.ReadString(msg.ReadableBytes, TextEncoding)
            _Debug?.DisposeResponse(connector, sMsg)
        End Sub

        ''' <summary>
        ''' 关闭观察者
        ''' </summary>
        Public Sub Dispose() Implements IDisposable.Dispose
            '_Debug = Nothing
            'TextEncoding = Nothing
            Return
        End Sub
    End Class

End Namespace
